bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BigtableVaneerSettingsFactory.java

```diff
@@ -0,0 +1,312 @@
+import com.google.api.gax.rpc.HeaderProvider;
+import com.google.auth.Credentials;
+import com.google.cloud.bigtable.admin.v2.BigtableTableAdminSettings;
+import com.google.cloud.bigtable.data.v2.BigtableDataSettings;
+import com.google.cloud.bigtable.data.v2.BigtableDataSettings.Builder;
+import com.google.cloud.bigtable.data.v2.models.InstanceName;
+import com.google.cloud.bigtable.data.v2.stub.BigtableStubSettings;
+import io.grpc.internal.GrpcUtil;
+import java.io.FileInputStream;
+import java.io.IOException;
+import org.threeten.bp.Duration;
+
+import static com.google.api.client.util.Preconditions.checkState;
+import static com.google.cloud.bigtable.config.CallOptionsConfig.SHORT_TIMEOUT_MS_DEFAULT;
+import static com.google.cloud.bigtable.config.CredentialOptions.getEnvJsonFile;
+import static org.threeten.bp.Duration.ofMillis;
+
+/**
+ * Static methods to convert an instance of {@link BigtableOptions} to a
+ * {@link BigtableDataSettings} or {@link BigtableTableAdminSettings} instance .
+ */
+public class BigtableVaneerSettingsFactory {
+
+  /** Constant <code>LOG</code> */
+  private static final Logger LOG = new Logger(BigtableVaneerSettingsFactory.class);
+
+  /**
+   * To create an instance of {@link BigtableDataSettings} from {@link BigtableOptions}.
+   *
+   * @param options a {@link BigtableOptions} object.
+   * @return a {@link BigtableDataSettings} object.
+   * @throws IOException if any.
+   */
+  public static BigtableDataSettings createBigtableDataSettings(final BigtableOptions options) throws IOException {
+    checkState(options.getProjectId() != null, "Project ID is required");
+    checkState(options.getInstanceId() != null, "Instance ID is required");
+    checkState(options.getRetryOptions().enableRetries(), "Disabling retries is not currently supported.");
+
+    BigtableDataSettings.Builder builder = BigtableDataSettings.newBuilder();
+    String endpoint = options.getDataHost() + ":" + options.getPort();
+
+    InstanceName instanceName = InstanceName.newBuilder().setProject(options.getProjectId())
+        .setInstance(options.getInstanceId()).build();
+    builder.setInstanceName(instanceName);
+    builder.setAppProfileId(options.getAppProfileId());
+
+    builder.setEndpoint(endpoint);
+
+    builder.setCredentialsProvider(buildCredentialProvider(options.getCredentialOptions()));
+
+    buildBulkMutationsSettings(builder, options);
+
+    buildCheckAndMutateRowSettings(builder, options.getCallOptionsConfig().getShortRpcTimeoutMs());
+
+    buildReadModifyWriteSettings(builder, options.getCallOptionsConfig().getShortRpcTimeoutMs());
+
+    buildReadRowsSettings(builder, options);
+
+    buildMutateRowSettings(builder, options);
+
+    buildSampleRowKeysSettings(builder, options);
+
+    String userAgent = BigtableVersionInfo.CORE_UESR_AGENT + "," + options.getUserAgent();
+    HeaderProvider headers = FixedHeaderProvider.create(GrpcUtil.USER_AGENT_KEY.name(), userAgent);
```

Comment:
Does this actually work or does grpc clobber it?

Reply:
I think It is getting translated to both the clients. 

I have added a simple test case for BigtableDataSettigns & BigtbaleTableAdminSettings with [ActualBigtable here](https://github.com/GoogleCloudPlatform/cloud-bigtable-client/pull/1973/files#diff-4acee08860ebdfaf1d83ea5d747b5508R140) (which we may remove as it might not needed here). 

So It created 32 ManagedChannels under channels pool when I run it in my local machine, Also It had the persisted user-agent on those channels. (attaching snapshot from debug console).
![Uploading DataClient-user-agent-verification.png…](BigtableDataClient-User-Agent)

Reply:
I think this needs to be verified on the server side and there should be a test to confirm this.  Please add a todo

Reply:
Can you add a comment about how gax-java has special handling for  the USER_AGENT_KEY:
https://github.com/googleapis/gax-java/blob/master/gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcHeaderInterceptor.java#L63-L74

Reply:
Added a separate test case for User-agent [TestUserAgent](https://github.com/GoogleCloudPlatform/cloud-bigtable-client/pull/1973/files#diff-8b4fb9399d066a371cb9b54a9a6d871cR59), verifying header with the help of HeaderServerInterceptor for [PlainText](https://github.com/GoogleCloudPlatform/cloud-bigtable-client/pull/1973/files#diff-8b4fb9399d066a371cb9b54a9a6d871cR91) as well as [TLS](https://github.com/GoogleCloudPlatform/cloud-bigtable-client/pull/1973/files#diff-8b4fb9399d066a371cb9b54a9a6d871cR122) Negotiation (_Using Client_trust, Server_Cert & Private key for the TLS user agent test case_).

Added [an inline comment here](https://github.com/GoogleCloudPlatform/cloud-bigtable-client/pull/1973/files#diff-6a492166d0005302ce3c542cdcf00ad5R283) about User-Agent handling.